HopsanGUI/Widgets/TextEditorWidget.cpp

```diff
@@ -589,7 +609,59 @@ void TextEditor::wheelEvent(QWheelEvent* event)
 //! @brief Updates list of auto complete words
 void TextEditor::updateAutoCompleteList()
 {
-    mpCompleter->setModel(new QStringListModel(gpTerminalWidget->mpHandler->getAutoCompleteWords(), mpCompleter));
+    if(mLanguage == HighlighterTypeEnum::Hcom) {
+        mpCompleter->setModel(new QStringListModel(gpTerminalWidget->mpHandler->getAutoCompleteWords(), mpCompleter));
+    }
+    else if(mLanguage == HighlighterTypeEnum::Cpp) {
+        QStringList lines = toPlainText().split("\n");
+
+        QStringList dataTypes = QStringList() << "size_t" << "double" << "int" << "SecondOrderTransferFunction" << "FirstOrderTransferFunction" << "Port";
```

Comment:
Feels like some of the "hopsanKeywordPatterns" should be reused here

Reply:
I added a static functions in CppHighlighter class for fetching the list with keyword patterns.